pkg/flow/tracing/tracing.go

```diff
@@ -14,7 +14,7 @@ import (
 	"go.opentelemetry.io/otel/exporters/otlp/otlptrace"
 	"go.opentelemetry.io/otel/sdk/resource"
 	tracesdk "go.opentelemetry.io/otel/sdk/trace"
-	semconv "go.opentelemetry.io/otel/semconv/v1.12.0"
+	semconv "go.opentelemetry.io/otel/semconv/v1.17.0"
```

Comment:
I think it should be safe. We only use this package for three things:
* `semconv.SchemaURL` has changed in [v1.12.0](https://github.com/open-telemetry/opentelemetry-go/blob/v1.15.0/semconv/v1.12.0/schema.go#L20) and [v1.17.0](https://github.com/open-telemetry/opentelemetry-go/blob/v1.16.0/semconv/v1.17.0/schema.go#L20). However, I don't think this will be a problem because `service.name` and `service.version` did not change.
* `semconv.ServiceNameKey` is the same in [v1.12.0](https://github.com/open-telemetry/opentelemetry-go/blob/v1.15.0/semconv/v1.12.0/resource.go#L919) and [v1.17.0](https://github.com/open-telemetry/opentelemetry-go/blob/v1.16.0/semconv/v1.17.0/resource.go#L1708)
* `semconv.ServiceVersionKey` is the same in [v1.12.0](https://github.com/open-telemetry/opentelemetry-go/blob/v1.15.0/semconv/v1.12.0/resource.go#L959) and [v1.17.0](https://github.com/open-telemetry/opentelemetry-go/blob/v1.16.0/semconv/v1.17.0/resource.go#L1759)